frame/bounties/src/lib.rs

```diff
@@ -653,12 +777,613 @@ decl_module! {
+					} = subbounty.status {
+						// Ensure block number is elapsed for
+						// processing the claim.
+						ensure!(
+							system::Module::<T>::block_number() >= *unlock_at,
+							Error::<T>::Premature,
+						);
+
+						// Make curator fee payment
+						let subbounty_account = Self::bounty_account_id(subbounty_id);
+						let balance = T::Currency::free_balance(&subbounty_account);
+						let fee = subbounty.fee.min(balance); // just to be safe
+						let payout = balance.saturating_sub(fee);
+
+						// unreserve the subcurator deposit
+						let _ = T::Currency::unreserve(
+							&subcurator,
+							subbounty.curator_deposit,
+						); // should not fail
+
+						// Make payout to subcurator
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							&subcurator,
+							fee,
+							AllowDeath,
+						); // should not fail
+
+						// Make payout to beneficiary
+						let _ = T::Currency::transfer(
+							&subbounty_account,
+							beneficiary,
+							payout,
+							AllowDeath,
+						); // should not fail
+
+						// Trigger the SubBountyClaimed event
+						Self::deposit_event(
+							Event::<T>::SubBountyClaimed(
+								bounty_id,
+								subbounty_id,
+								payout,
+								beneficiary.clone(),
+							)
+						);
+
+						// Remove the subbounty from bounty active subbouty list
+						Bounties::<T>::mutate_exists(
+							bounty_id,
+							|maybe_bounty| -> DispatchResult {
+								// Remove the subbounty index from parent bounty
+								// active list.
+								if let Some(bounty) = maybe_bounty.as_mut() {
+									bounty.active_subbounty_count -= 1;
+								}
+								Ok(())
+							}
+						)?;
+						// Remove the subbounty description
+						BountyDescriptions::remove(subbounty_id);
+						// Remove the subbounty instance
+						// from DB
+						*maybe_subbounty = None;
+						Ok(())
+					} else {
+						Err(Error::<T>::UnexpectedStatus.into())
+					}
+				}
+			)?;
+		}
+
+		/// Cancel a proposed or active subbounty.
+		/// Subbounty account funds are transferred to parent bounty account.
+		/// the subcurator deposit may be unreserved if possible.
+		///
+		/// The dispatch origin for this call must be
+		/// either master curator of this subbounty or `T::RejectOrigin`.
+		///
+		/// If state of subbounty is `Active`,
+		/// subcurator deposit is unreserved.
+		///
+		/// If state of subbounty is `PendingPayout`,
+		/// call fails & returns PendingPayout error.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Instance of subbounty is deallocated from DB
+		/// on successful call completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = <T as Config>::WeightInfo::close_subbounty()]
+		fn close_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) -> DispatchResultWithPostInfo {
```

Comment:
we don't need refund weight so no need to have return type
```suggestion
		) {
```

Reply:
Got it, Changes Done  :+1: